subspace_model/logic.py

```diff
@@ -324,106 +310,143 @@ def p_storage_fees(
-        "holders_balance": -eff_total_storage_fees,
-        "storage_fee_volume": eff_total_storage_fees,
+        # Fee Calculation
+        "free_space": free_space,
+        "transaction_byte_fee": transaction_byte_fee,
+        "extrinsic_length_in_bytes": extrinsic_length_in_bytes,
+        "storage_fee_volume": eff_storage_fee_volume,
+        # Reward Distribution
+        "storage_fees_to_farmers": storage_fees_to_farmers,
+        "farmers_balance": storage_fees_to_farmers,
+        "storage_fees_to_fund": storage_fees_to_fund,
+        "fund_balance": storage_fees_to_fund,
+        "holders_balance": -eff_storage_fee_volume,
     }
 
 
 def p_compute_fees(
     params: SubspaceModelParams, _2, _3, state: SubspaceModelState
 ) -> PolicyOutput:
     """
-    HACK: If holders balance is insufficient, then the amount of paid fees
+    Calculate compute fees.
+
+    If holders balance is insufficient, then the amount of paid fees
     will be lower even though the transactions still go through.
+
+    Reference: https://subspacelabs.notion.site/Fees-Rewards-Specification-WIP-1b835c7684a940f188920802ca6791f2#4d2c4f4b69a94fcca49a7fcaca7563cc
     """
 
-    tx_compute_weight = (
+    weight_to_fee = params["weight_to_fee"]
+    max_normal_weight = 0.75 * BLOCK_WEIGHT_FOR_2_SEC
+    max_bundle_weight = state["max_bundle_weight"]
+    target_block_fullness = state["target_block_fullness"]
+    block_weight_utilization = state["block_utilization"]
+    adjustment_variable = state["adjustment_variable"]
+    priority_fee_volume = state["average_priority_fee"]
+
+    target_block_delta = target_block_fullness - block_weight_utilization
+
+    targeted_adjustment_parameter = (
+        1
+        + adjustment_variable * target_block_delta
+        + adjustment_variable**2 * target_block_delta**2 / 2
+    )
+
+    prev_compute_fee_multiplier = state["compute_fee_multiplier"]
+    compute_fee_multiplier = targeted_adjustment_parameter * prev_compute_fee_multiplier
+
+    # Caculate compute weight
+    tx_compute_weight: ComputeWeights = (
         state["average_compute_weight_per_tx"] * state["transaction_count"]
     )
     bundles_compute_weight = (
         state["average_compute_weight_per_bundle"] * state["bundle_count"]
     )
 
     total_compute_weights: ComputeWeights = tx_compute_weight + bundles_compute_weight
-    base_fees: Credits = (
-        state["average_base_fee"] * total_compute_weights * SHANNON_IN_CREDITS
-    )
-    priority_fees: Credits = (
-        state["average_priority_fee"] * total_compute_weights * SHANNON_IN_CREDITS
-    )
 
-    total_fees = base_fees + priority_fees
-    eff_total_fees = min(total_fees, state["holders_balance"])  # HACK
-    eff_scale = eff_total_fees / total_fees
+    # Calculate compute fee volume
+    compute_fee_volume = max(
+        (
+            compute_fee_multiplier * weight_to_fee * bundles_compute_weight
+            + priority_fee_volume
+        ),
+        1,
```

Comment:
Yes minimum 1 SSC.

It sets a minimum compute_fee_volume of 1 to prevent divide by 0 in the eff_scale calculation. 

The straight line of 1 might be because of the miscalculation using bundles_compute_weight instead of total_compute_weight as you pointed out earlier.

Reply:
Oh okay I see. 1 SSC is expensive! If you consider that 1 picosecond (10^-12) costs 1 Shannon (current `WeightToFee`) than a full block of 1.5 seconds of compute is 1.5*10^12 Shannons, and you would need another ~10^6 seconds to get to 1 SSC. Storage fees are usually below that ~10^9-10^13 Shannon too. 
If you need a 1 there to prevent zeroing , it should be 1 shannon.

Reply:
Awww thanks @dariolina that's much more clear now. 1 shannon makes sense for the minimum.